service/service_test.go

```diff
@@ -154,6 +162,22 @@ func assertMetricsPrefix(t *testing.T, prefix string, metricsPort uint16) {
 			"expected prefix %q but string starts with %q",
 			prefix,
 			s[:len(prefix)+1]+"...")
+
+		for _, mandatoryLabel := range mandatoryLabels {
+			i := strings.Index(s, mandatoryLabel)
+			require.True(
+				t,
+				i > 0,
+				"mandatory label %q not found on metric: %s",
+				mandatoryLabel,
+				s)
+			require.True(
+				t,
+				s[i-1] == '{' || s[i-1] == ',',
```

Comment:
This is an oddly specific and error prone check. Is there any way to make it less hacky / easier to read?

We should also check for 
```go
s[i+len(mandatoryLabel)] == ',' || s[i+len(mandatoryLabel)] == '}'
```
for completeness, see metric example:
```
otelcol_exporter_send_failed_metric_points{exporter="logging",service_instance_id="15d68785-b17f-4a8c-8d24-638edfc38726"} 0
```

I would suggest writing a helper function that extracts all labels instead, i.e. 
```go
func extractPrometheusLabels(s string) map[string]string {
  ...
}
```
or use Prometheus parser if that's possible.

Reply:
Looks like you can use https://godoc.org/github.com/prometheus/common/expfmt#TextParser.TextToMetricFamilies to parse the Prometheus text format.